src/operations/connect.ts

```diff
@@ -197,6 +197,11 @@ export function connect(
     throw new Error('no callback function provided');
   }
 
+  // Has a connection already been established?
+  if (mongoClient.topology && mongoClient.topology.isConnected()) {
+    throw new Error(`'connect' cannot be called when already connected`);
```

Comment:
Let's not use plain `Error` objects, instead prefer a `TypeError` for simple type mismatches and `MongoError` or one of its subclasses for the rest. In this case I think a `MongoError` would be best. (I know the code right above this uses an `Error`, but let's be the change we want to see in the world right?)